### DISCUSSION TITLE
Cookies and Headers in NextJS 13 App Dir

### DISCUSSION THREAD

**Discussion by jleclanche** [vercel/community#1132] [Help] [answered]
The docs are very clear, these are read-only:

- https://beta.nextjs.org/docs/api-reference/cookies
- https://beta.nextjs.org/docs/api-reference/headers#headers

> Note: The Next.js team at Vercel is working on adding the ability to set cookies in addition to the cookies function.

Where is this work being done? Any way to help with it? Setting/deleting a cookie is pretty basic functionality so I'm blocked on using NextJS 13 for a product I'm working on right now. And I found this out late enough that I am already pretty deeply locked in to the layout for the system in question so I can't just swap over that one page =/

I would love to contribute code as necessary, but I'm having trouble even finding documentation about this - if it's already being worked on, then I'd like to not step on any toes and contribute in a useful manner. But there doesn't seem to be an issue about this.

**Comment by leerob** [comment, marked as answer]
The `app` directory is still in beta and not recommended for production yet. Next.js 13 and the `pages` directory are stable, so I would recommend that if you are looking to put this product into production.

You can currently workaround this by using Middleware to set cookies, but it's not very ergonomic, so we're working on a better solution (requires a public RFC).

**Comment by aamirrashid74040** [comment]
I am importing cookies from next/header but it giving me error in client side component 
How can i get token from cookie in client side conponent

**Reply by spirituelle** [reply]
to get it you can use the server actions. create a new file called actions and put the function that save your cookie there
